codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -272,10 +272,10 @@ private boolean allowNotificationEvent() {
         public ChannelPromise setSuccess(Void result) {
             if (allowNotificationEvent()) {
                 ++successfulCount;
-                if (successfulCount == expectedCount && doneAllocating) {
-                    promise.setSuccess(result);
-                    return super.setSuccess(result);
-                }
+            }
+            if (successfulCount == expectedCount && doneAllocating && !isDone()) {
```

Comment:
I'm not sure I understand this change.  The `allowNotificationEvent()` was meant to detect if we are still expecting success/failure events.  When would it make sense to notify the promise potentially after we have received all the events we expected (seems like we added a `!isDone()` to compensate for this?).